apsapi/__init__.py

```diff
@@ -0,0 +1,122 @@
+#!/usr/bin/python
+
+import json
+import urllib2
+import httplib
+import ssl
+from urlparse import urljoin
+
+# compatibility with python 2.7.9+
+# see https://www.python.org/dev/peps/pep-0476/
+_PYTHON_2_7_9_COMPAT = False
+if '_create_unverified_context' in dir(ssl):
+    _PYTHON_2_7_9_COMPAT = True
+
+
+class HTTPSClientAuthHandler(urllib2.HTTPSHandler):
+    def __init__(self, key, cert):
+        context = None
+        if _PYTHON_2_7_9_COMPAT:
+            context = ssl._create_unverified_context()
+        urllib2.HTTPSHandler.__init__(self, context=context)
+
+        self.key = key
+        self.cert = cert
+
+    def https_open(self, req):
+        # Rather than pass in a reference to a connection class, we pass in
+        # a reference to a function which, for all intents and purposes,
+        # will behave as a constructor
+        return self.do_open(self.get_connection, req, context=self._context)
+
+    def get_connection(self, host, context=None):
+        return httplib.HTTPSConnection(host, key_file=self.key, cert_file=self.cert, context=context)
+
+
+class API(object):
+
+    class APSExcStruct:
+        def __init__(self):
+            self.code = 0
+            self.type = ''
+            self.message = ''
+
+    def __init__(self, url):
+        """
+        Takes something like the following argument as input:
+        url = 'https://a.bvt.aps.sw.ru:6308'
+        """
+        self.url = url
+
+    def call(self, verb, path, headers=None, data=None, cert=None, rheaders=None):
+        data = json.dumps(data)
+
+        # url = self.url + path
+        url = urljoin(self.url, path)
+        if not headers:
+            headers = dict()
+
+        req = urllib2.Request(url, headers=headers, data=data)
+        req.get_method = lambda: verb
+
+        resp = None
+        try:
+            if cert:
+                opener = urllib2.build_opener(HTTPSClientAuthHandler(cert, cert))
+                resp = opener.open(req)
+            else:
+                if _PYTHON_2_7_9_COMPAT:
```

Comment:
use logic like in `HTTPSClientAuthHandler.__init__` with `context` as `None` by default